Makefile

```diff
@@ -209,9 +211,10 @@ vendor-in-container:
 
 .PHONY: vendor
 vendor:
-	GO111MODULE=on $(GO) mod tidy
-	GO111MODULE=on $(GO) mod vendor
-	GO111MODULE=on $(GO) mod verify
+	$(GO) mod tidy
+	$(GO) mod vendor
+	$(GO) mod verify
+	if test -n "$(strip $(shell go env GOTOOLCHAIN))"; then go mod edit -toolchain none ; fi
```

Comment:
I rather not remove the toolchain line and have behave consistently across all our projects (https://github.com/containers/skopeo/pull/2297, https://github.com/containers/common/pull/1963 and others ...)
I don't think this is wrong, the end result is the same as far as I can tell but consistency is key IMO

cc @mtrmac

Reply:
Whatever changes we do here impact the vendoring validation tests.  I'm not averse to splitting it out, but it has effects on things that come after it.

Reply:
https://go.dev/ref/mod:
> For reproducibility, the go command writes its own toolchain name in a toolchain line any time it is updating the go version in the go.mod file 

and IIRC that’s the _only_ time that happens. So, in c/image and Skopeo we have a `toolchain 1.21.0` line; then `go get` does not trigger any updates to 1.21.x as long as we stay on Go 1.21.

---

A possible unanticipated downside of the `toolchain 1.21.0` approach is that Renovate files PRs just to update the `toolchain`, as in https://github.com/containers/image/pull/2382 .

I don’t think we know for sure what Renovate does with Go ≥ 1.21 if there is no `toolchain` directive at all, AFAIK all updates to 1.21 were using the `toolchain 1.21.0` approach so far.

Reply:
… I think what I’m saying here is that IIRC we haven’t considered the approach of just editing out the `toolchain` dependency.

And it might actually be a pretty good solution; if nothing else, reflects our intent more precisely than a `1.21.0` value.

… but also, I… don’t personally want to be spending much more time on `toolchain`. If this approach is better, I’ll happily help switch other repos, but I’d rather not invest much time evaluating it myself.